From 23415232026b35dfe771788cb33c5c5fce647282 Mon Sep 17 00:00:00 2001
From: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
Date: Thu, 20 May 2021 00:20:15 +0800
Subject: [PATCH 110/122] igc: Increase timeout value for Speed 100/1000/2500

As the cycle time is set to maximum of 1s, the TX Hang timeout need to
be increase to avoid possible TX Hangs caused by using long Qbv cycles.

Signed-off-by: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
---
 drivers/net/ethernet/intel/igc/igc_main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/intel/igc/igc_main.c b/drivers/net/ethernet/intel/igc/igc_main.c
index e38cbbf3340c..e5ee7340606f 100644
--- a/drivers/net/ethernet/intel/igc/igc_main.c
+++ b/drivers/net/ethernet/intel/igc/igc_main.c
@@ -5018,7 +5018,10 @@ static void igc_watchdog_task(struct work_struct *work)
 				adapter->tx_timeout_factor = 14;
 				break;
 			case SPEED_100:
+			case SPEED_1000:
+			case SPEED_2500:
 				/* maybe add some timeout factor ? */
+				adapter->tx_timeout_factor = 7;
 				break;
 			}
 
-- 
2.17.1

